apps/meteor/server/methods/deleteUser.ts

```diff
@@ -15,45 +15,56 @@ declare module '@rocket.chat/ddp-client' {
 	}
 }
 
-Meteor.methods<ServerMethods>({
-	async deleteUser(userId, confirmRelinquish = false) {
-		check(userId, String);
-		const uid = Meteor.userId();
-		if (!uid || (await hasPermissionAsync(uid, 'delete-user')) !== true) {
-			throw new Meteor.Error('error-not-allowed', 'Not allowed', {
-				method: 'deleteUser',
-			});
-		}
+export const executeDeleteUser = async (fromUserId: IUser['_id'], userId: IUser['_id'], confirmRelinquish = false): Promise<boolean> => {
+	if ((await hasPermissionAsync(fromUserId, 'delete-user')) !== true) {
+		throw new Meteor.Error('error-not-allowed', 'Not allowed', {
+			method: 'deleteUser',
+		});
+	}
 
-		const user = await Users.findOneById(userId);
-		if (!user) {
-			throw new Meteor.Error('error-invalid-user', 'Invalid user to delete', {
-				method: 'deleteUser',
-			});
-		}
+	const user = await Users.findOneById(userId);
+	if (!user) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user to delete', {
+			method: 'deleteUser',
+		});
+	}
 
-		if (user.type === 'app') {
-			throw new Meteor.Error('error-cannot-delete-app-user', 'Deleting app user is not allowed', {
-				method: 'deleteUser',
-			});
-		}
+	if (user.type === 'app') {
+		throw new Meteor.Error('error-cannot-delete-app-user', 'Deleting app user is not allowed', {
+			method: 'deleteUser',
+		});
+	}
+
+	const adminCount = await Users.countDocuments({ roles: 'admin' });
+
+	const userIsAdmin = user.roles?.indexOf('admin') > -1;
 
-		const adminCount = await Users.countDocuments({ roles: 'admin' });
+	if (adminCount === 1 && userIsAdmin) {
+		throw new Meteor.Error('error-action-not-allowed', 'Leaving the app without admins is not allowed', {
+			method: 'deleteUser',
+			action: 'Remove_last_admin',
+		});
+	}
+
+	await deleteUser(userId, confirmRelinquish, fromUserId);
 
-		const userIsAdmin = user.roles?.indexOf('admin') > -1;
+	// App IPostUserDeleted event hook
+	await Apps.self?.triggerEvent(AppEvents.IPostUserDeleted, { user, performedBy: await Users.findOneById(fromUserId) });
```

Comment:
this may need to be moved to the `deleteUser` function

Reply:
I will not change that in this PR. I will add a TODO, because probably there was a reason for the trigger to be here instead of on that function 😨